routers/repo/pull.go

```diff
@@ -253,40 +253,30 @@ func setMergeTarget(ctx *context.Context, pull *models.PullRequest) {
 }
 
 // PrepareMergedViewPullInfo show meta information for a merged pull request view page
-func PrepareMergedViewPullInfo(ctx *context.Context, issue *models.Issue) {
+func PrepareMergedViewPullInfo(ctx *context.Context, issue *models.Issue) *git.PullRequestInfo {
 	pull := issue.PullRequest
 
-	var err error
-	if err = pull.GetHeadRepo(); err != nil {
-		ctx.ServerError("GetHeadRepo", err)
-		return
-	}
-
 	setMergeTarget(ctx, pull)
 	ctx.Data["HasMerged"] = true
 
-	mergedCommit, err := ctx.Repo.GitRepo.GetCommit(pull.MergedCommitID)
-	if err != nil {
-		ctx.ServerError("GetCommit", err)
-		return
-	}
-	// the ID of the last commit in the PR (not including the merge commit)
-	endCommitID, err := mergedCommit.ParentID(mergedCommit.ParentCount() - 1)
-	if err != nil {
-		ctx.ServerError("ParentID", err)
-		return
-	}
+	prInfo, err := ctx.Repo.GitRepo.GetPullRequestInfo(ctx.Repo.Repository.RepoPath(),
+		pull.MergeBase, pull.GetGitRefName())
 
-	ctx.Data["NumCommits"], err = ctx.Repo.GitRepo.CommitsCountBetween(pull.MergeBase, endCommitID.String())
-	if err != nil {
-		ctx.ServerError("Repo.GitRepo.CommitsCountBetween", err)
-		return
-	}
-	ctx.Data["NumFiles"], err = ctx.Repo.GitRepo.FilesCountBetween(pull.MergeBase, endCommitID.String())
 	if err != nil {
-		ctx.ServerError("Repo.GitRepo.FilesCountBetween", err)
-		return
+		if strings.Contains(err.Error(), "fatal: Not a valid object name") {
+			ctx.Data["IsPullReuqestBroken"] = true
```

Comment:
Looks like there's a typo here (`Reuqest`), although it is spread in the entire codebase. Can you change it now or do you think it's better to make another PR after this one is merged?

Reply:
Nice catch, IMO a separate PR would be better

Reply:
I also think it should be separate PR as this one fixes different issue and that change would make this one harder to review